pkg/app/pipedv1/plugin/kubernetes/provider/resource.go

```diff
@@ -87,6 +87,9 @@ func MakeResourceKey(obj *unstructured.Unstructured) ResourceKey {
 		Namespace:  obj.GetNamespace(),
 		Name:       obj.GetName(),
 	}
+	if k.Namespace == "" {
+		k.Namespace = DefaultNamespace
+	}
```

Comment:
There was an incompatibility between the current Kubernetes platform provider and the Kubernetes plugin, so I fixed this.

Reply:
Thank you for commenting. I reverted this change and fixed the assertion.
We can write the e2e test now, so I want to write tests to prevent bugs.

https://github.com/pipe-cd/pipecd/pull/5431/commits/057b9f55222d546bf363e7f252ccfb27f64f59f1
https://github.com/pipe-cd/pipecd/pull/5431/commits/e030be1f13b4a06ede8407fb2fe45bd146d710d9